xbmc/addons/InputStream.cpp

```diff
@@ -91,21 +94,22 @@ bool CInputStream::Supports(CFileItem &fileitem)
   }
 
   match = false;
+
   for (auto &path : m_pathList)
   {
     if (path.empty())
       continue;
 
-    if (fileitem.GetPath().compare(0, path.length(), path) == 0)
+    CRegExp r(true, CRegExp::asciiOnly, path.c_str());
```

Comment:
please elaborate. I don't see any advantage in replacing a single line of code by 3 and introducing regex here. what exactly are you trying to achieve?